src/Mollie/WC/Payment/Object.php

```diff
@@ -425,6 +425,12 @@ public function getMollieCustomerIdFromPaymentObject() {
 
 	}
 
+	public function cancelOrderOnManualUpdate($orderId, $status){
+        add_filter( Mollie_WC_Plugin::PLUGIN_ID . '_order_status_cancelled', function(){
+            return Mollie_WC_Gateway_Abstract::STATUS_CANCELLED;
+        } );
+    }
```

Comment:
May work, but I'm wondering why this problem happened at all? I looked at the Mollie plugin code and didn't found anything able to cause the problem, except for the incoming webhook. Can it be that when the order cancelled manually, it cancelled then in Mollie by calling their API (makes sense), then Mollie triggers webhook because the order was cancelled there and then the `onWebhookCancelled` method sets order status to `Pending` or any other status selected in settings? 

I'm asking because if the problem caused in the way I described, then this shouldn't work.

Probably, it makes sense to not change the order status when not needed (I assuming it's Mollie plugin changes the order status after cancelling), instead of filtering it back later?